src/main/java/betterquesting/questing/QuestInstance.java

```diff
@@ -20,562 +20,519 @@
-	
-	private final HashMap<UUID, NBTTagCompound> completeUsers = new HashMap<>();
+public class QuestInstance implements IQuest {
+    private final TaskStorage tasks = new TaskStorage();
+    private final RewardStorage rewards = new RewardStorage();
+
+    private final HashMap<UUID, NBTTagCompound> completeUsers = new HashMap<>();
     private int[] preRequisites = new int[0];
-	
-	private final PropertyContainer qInfo = new PropertyContainer();
-	
-	public QuestInstance()
-	{
-		this.setupProps();
-	}
-	
-	private void setupProps()
-	{
-		setupValue(NativeProps.NAME, "New Quest");
-		setupValue(NativeProps.DESC, "No Description");
-		
-		setupValue(NativeProps.ICON, new BigItemStack(Items.NETHER_STAR));
-		
-		setupValue(NativeProps.SOUND_COMPLETE);
-		setupValue(NativeProps.SOUND_UPDATE);
-		//setupValue(NativeProps.SOUND_UNLOCK);
-		
-		setupValue(NativeProps.LOGIC_QUEST, EnumLogic.AND);
-		setupValue(NativeProps.LOGIC_TASK, EnumLogic.AND);
-		
-		setupValue(NativeProps.REPEAT_TIME, -1);
-		setupValue(NativeProps.REPEAT_REL, true);
-		setupValue(NativeProps.LOCKED_PROGRESS, false);
-		setupValue(NativeProps.AUTO_CLAIM, false);
-		setupValue(NativeProps.SILENT, false);
-		setupValue(NativeProps.MAIN, false);
-		setupValue(NativeProps.GLOBAL_SHARE, false);
-		setupValue(NativeProps.SIMULTANEOUS, false);
-		setupValue(NativeProps.VISIBILITY, NativeProps.VISIBILITY.getDefault());
-	}
-	
-	private <T> void setupValue(IPropertyType<T> prop)
-	{
-		this.setupValue(prop, prop.getDefault());
-	}
-	
-	private <T> void setupValue(IPropertyType<T> prop, T def)
-	{
-		qInfo.setProperty(prop, qInfo.getProperty(prop, def));
-	}
-	
-	@Override
-	public void update(EntityPlayer player)
-	{
-		UUID playerID = QuestingAPI.getQuestingUUID(player);
-		
+    private final TIntObjectMap<RequirementType> prereqTypes = new TIntObjectHashMap<>();
+
+    private final PropertyContainer qInfo = new PropertyContainer();
+
+    public QuestInstance() {
+        this.setupProps();
+    }
+
+    private void setupProps() {
+        setupValue(NativeProps.NAME, "New Quest");
+        setupValue(NativeProps.DESC, "No Description");
+
+        setupValue(NativeProps.ICON, new BigItemStack(Items.NETHER_STAR));
+
+        setupValue(NativeProps.SOUND_COMPLETE);
+        setupValue(NativeProps.SOUND_UPDATE);
+        //setupValue(NativeProps.SOUND_UNLOCK);
+
+        setupValue(NativeProps.LOGIC_QUEST, EnumLogic.AND);
+        setupValue(NativeProps.LOGIC_TASK, EnumLogic.AND);
+
+        setupValue(NativeProps.REPEAT_TIME, -1);
+        setupValue(NativeProps.REPEAT_REL, true);
+        setupValue(NativeProps.LOCKED_PROGRESS, false);
+        setupValue(NativeProps.AUTO_CLAIM, false);
+        setupValue(NativeProps.SILENT, false);
+        setupValue(NativeProps.MAIN, false);
+        setupValue(NativeProps.GLOBAL_SHARE, false);
+        setupValue(NativeProps.SIMULTANEOUS, false);
+        setupValue(NativeProps.VISIBILITY, EnumQuestVisibility.NORMAL);
```

Comment:
This change for visibility should not be done, as it undoes the config option that was added for the default quest visibility.